docs/update-from-client-side-helpers-to-server-side.md

```diff
@@ -0,0 +1,229 @@
+
+          // Recoverable state, per:
+          // https://developer.paypal.com/docs/checkout/integration-features/funding-failure/
+          if (errorDetail && errorDetail.issue === 'INSTRUMENT_DECLINED') {
+              return actions.restart(); 
+          }
+
+          if (errorDetail) {
+              console.error('Sorry, your transaction could not be processed.', errorDetail);
+          }
+
+          // Successful capture! For demo purposes:
+          console.log('Capture result', orderData, JSON.stringify(orderData, null, 2));
+      })
+      .catch(function (err) {
+          // Failed capture
+      });
+}
+```
+
+## Benefits of using a Server-side Integration
+
+* Secure Authorization with PayPal's APIs: Use a CLIENT_ID and CLIENT_SECRET to securely consume PayPal's APIs from your server-side code. The CLIENT_SECRET is only known by your server-side code and uniquely identifies your application.
+* Secure Order Creation: Keep sensitive data, such as order amount, on the server to prevent tampering by outside actors.
+
+## Best Practices for JS SDK Server-Side Integrations:
+
+1. CLIENT_SECRET should never be checked into git. We recommend passing this sensitive value to the web server at runtime as an environment variable. It's common to use a .env file that is ignored by git to load sensitive values like this secret.
+
+To securely store your credentials in your application code, we recommend using a .env file. You can find the relevant libraries for Node.js, PHP, and Python at the following links:
+  - Node.js: https://github.com/motdotla/dotenv
+  - PHP: https://github.com/vlucas/phpdotenv
+  - Python: https://github.com/theskumar/python-dotenv
+
+2. The client credentials auth token returned by `/v1/oauth2/token` api endpoint should never be passed to the browser. Keep this value in memory on the server-side and use it as the Authorization header for all other api calls.  
+Here's an example API call in Node.js that uses the client credentials auth token as the Authorization header for all other API calls:
+```
+const encodedClientCredentials = Buffer.from(`${client}:${secret}`).toString("base64");
+
+const response = await fetch(`${apiBaseUrl}/v1/oauth2/token`, {
+    method: "POST",
+    body: "grant_type=client_credentials",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/x-www-form-urlencoded",
+      "Accept-Language": "en_US",
+      Authorization: `Basic ${encodedClientCredentials}`,
+    });
+    
+const data = await response.json();
+
+// do not expose "data.access_token" to the browser
+```
+
+3. Create API endpoints to wrap the PayPal APIs. These API endpoints should include error handling.
+  * for errors, include the response body in the API's error response to provide helpful information about the error. For example, when capturing an order, an 'INSTRUMENT_DECLINED' error might be returned in the response body.
```

Comment:
Looks like both "API's" and "APIs" are used throughout the doc.  I think it would be good to settle on one or the other for consistency.